synapse/handlers/message.py

```diff
@@ -98,10 +133,19 @@ def get_state_events(self, user_id, room_id, is_guest=False):
         )
 
         if membership == Membership.JOIN:
-            room_state = yield self.state.get_current_state(room_id)
+            if at_event:
+                room_state = yield self.store.get_state_for_events(
+                    [at_event], types, filtered_types=filtered_types
```

Comment:
gentle reminder to use trailing commas in this situation. don't mind if you want to leave these now they are done.